sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -259,6 +259,29 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
         builder.append("]")
         builder.build()
       })
+    case StructType(fields) =>
+      buildCast[InternalRow](_, row => {
+        val builder = new UTF8StringBuilder
+        builder.append("[")
```

Comment:
Are there any reasons for using the same brackets `[]` for Struct, Map and Array?

How about `[]` for arrays, `{}` for structs and `<>` for maps or somehow distinguish them. At least, it is not consistent with `toHiveString` which prints `{}` for structs and maps.

Reply:
I think we have no strong reason. In this PR, I just followed the existing format. Looks like we can change it to `{` for consistency. cc: @cloud-fan

Reply:
I'm OK with the new format.

Reply:
Here is the PR https://github.com/apache/spark/pull/29308 . Please, review it.